### DISCUSSION TITLE
Can I use LogTo RBAC to implement resource-scoped permissions?

### DISCUSSION THREAD

**Discussion by andy250** [logto-io/logto#5488] [Q&A]
Building a small SaaS with LogTo + Blazor Server (.net8). The basic setup worked for me. I will have multiple Organizations, each with multiple users and multiple projects (dynamically created over time). I need to set permissions per project level. So user A requires Read on project X and Read+Write on project Y. Is this doable in LogTo? Can I simply add permissions that include the project ID as part of the permission name? E.g. permission:read:project:x,  permission:read:project:y? Is there a limit on number of permission in LogTo? (the number of projects will grow over time).

**Comment by simeng-li** [comment]
Hey Adny, nice to meet U. As you mentioned currently all the RBAC permissions are defined at the resource level in Logto. So what exactly do you mean by "projects" in your system? Are these individual applications, distinct services, or some other resource entities?

**Comment by andy250** [comment]
Hey @simeng-li. In my system projects are resource entities. If my product was github, then a single project would equal to github's single repository (which belongs to a user, has read/write permissions specified per user etc).

**Comment by simeng-li** [comment]
I see.  Sounds totally doable to me. 

Feel free to set as many permissions as needed – there are no restrictions on the number.  You can certainly use `permission:read:project:x` format to define those permissions.  

Explore either the [resource-based RBAC](https://docs.logto.io/docs/recipes/rbac/) or leverage [organization permissions](https://docs.logto.io/docs/recipes/organizations/understand-how-it-works/#organization-permission) based on your preferences.  The key distinction lies in whether you seek permission during an API request for resource access or when attempting to access the resources of an organization. It really depends on your use case, whichever fits your resource structure. 

P.S. probably you would like to check our [organization template](https://docs.logto.io/docs/recipes/organizations/configuration/#configure-organization-template) feature which allows you to define roles template across multiple organizations.  e.g. define roles based on different projects you have. So you may easity assign them cross orgs.

**Comment by JPustkuchen** [comment]
I agree this is quite a typical requirement for multi-tenant, multi-project applications. We should try to find out, if it works well with the existing functionality or if it needs more specific targeting in API and UI.

Maybe RBAC isn't the perfect choice for this, instead you may need PBAC or ReBAC for such, even more complex scenarios.
